File: open-days-backend/src/main/java/com/sapientia/open/days/backend/ui/model/BaseError.java
package com.sapientia.open.days.backend.ui.model;

/**
 * A basic model to return an error to the client. The BaseException uses this as return type.
 */
@SuppressWarnings("unused")
public class BaseError {

	private int errorCode;
	private String errorMessage;

	public BaseError() {
	}

	public BaseError(int errorCode, String errorMessage) {
		this.errorCode = errorCode;
		this.errorMessage = errorMessage;
	}

	public int getErrorCode() {
		return errorCode;
	}

	public String getErrorMessage() {
		return errorMessage;
	}

	public void setErrorCode(int errorCode) {
		this.errorCode = errorCode;
	}

	public void setErrorMessage(String errorMessage) {
		this.errorMessage = errorMessage;
	}
}

File: open-days-frontend/lib/models/base_error.dart
/// A basic model to represent an error from the server.
class BaseError {
  int errorCode;
  String errorMessage;

  BaseError({
    this.errorCode = -1,
    this.errorMessage = '',
  });

  factory BaseError.fromJson(Map<String, dynamic> json) {
    return BaseError(
      errorCode: json['errorCode'] ?? '',
      errorMessage: json['errorMessage'] ?? '',
    );
  }
}
